api/azure/api.go

```diff
@@ -62,10 +74,40 @@ func (a apiImpl) HandleCheckRunEvent(checkRun *github.CheckRunEvent) (bool, erro
 	return handleCheckRunEvent(a, shared.NewAppEngineAPI(a.ctx), checkRun)
 }
 
+func (a apiImpl) GetAzureBuildURL(owner, repo string, buildID int64) string {
+	// https://docs.microsoft.com/en-us/rest/api/azure/devops/build/builds/get?view=azure-devops-rest-4.1#build
+	return fmt.Sprintf(
+		"https://dev.azure.com/%s/%s/_apis/build/builds/%v", owner, repo, buildID)
+}
+
 func (a apiImpl) GetAzureArtifactsURL(owner, repo string, buildID int64) string {
 	return fmt.Sprintf(
 		"https://dev.azure.com/%s/%s/_apis/build/builds/%v/artifacts",
 		owner,
 		repo,
 		buildID)
 }
+
+func (a apiImpl) IsMasterBranch(owner, repo string, buildID int64) bool {
+	buildURL := a.GetAzureBuildURL(owner, repo, buildID)
+	client := shared.NewAppEngineAPI(a.ctx).GetHTTPClient()
+	log := shared.GetLogger(a.ctx)
+	resp, err := client.Get(buildURL)
+	if err != nil {
+		log.Errorf("Failed to fetch build: %s", err.Error())
+		return false
+	}
+	defer resp.Body.Close()
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Errorf("Failed to read request response: %s", err.Error())
+		return false
+	}
+	var build AzureBuild
+	if err := json.Unmarshal(data, &build); err != nil {
+		log.Errorf("Failed to unmarshal request response: %s", err.Error())
+		return false
+	}
+	log.Debugf("Source branch: %s", build.TriggerInfo.SourceBranch)
+	return build.TriggerInfo.SourceBranch == "master"
```

Comment:
This will fail to add the master label for builds triggered from epochs/* branches. It's a bit ugly to know about branch topology here, but the only other option is to have a checkout.

Reply:
Can you give a specific build ID for this case?